google/cloud/storage/client.h

```diff
@@ -829,7 +941,13 @@ class Client {
    *     `IfMetagenerationNotMatch`, `PredefinedAcl`,
    *     `Projection`, and `UserProject`.
    *
-   * @throw std::runtime_error if the operation fails.
+   * @throw std::runtime_error if there is a permanent failure, or if there were
+   *     more transient failures than allowed by the current retry policy.
+   *
+   * @par Idempotency
+   * Like other operations that mutate the state of GCS, this operation is only
+   * idempotent if restricted by pre-conditions, in this case,
+   * `IfMetagenerationMatch`.
```

Comment:
I would suspect a IfGenerationMatch as well to check if it's the same blob data?

Reply:
I think a similar reasoning to `CopyObject` applies here: setting `IfGenerationMatch` by itself would not guarantee idempotency, setting `IfMetagenerationMatch` does because this function changes the metageneration number.  Setting both is unnecessary because well, it is already idempotent with `IfMetagenerationMatch`.